src/game_api/state_structs.hpp

```diff
@@ -522,6 +522,22 @@ class LogicSunChallenge : public Logic
     virtual ~LogicSunChallenge() = 0;
 };
 
+class MagmamanSpawnPosition
+{
+  public:
+    uint32_t x;
+    uint32_t y;
+    uint32_t timer;
+};
+
+class LogicMagmamanSpawn : public Logic
+{
+  public:
+    std::vector<MagmamanSpawnPosition> magmaman_positions;
+
+    virtual ~LogicMagmamanSpawn() = 0;
```

Comment:
can you remove this virtual, it's not correct, not sure why all the logic have a separate destructor written, it's not there in the game, will change the others in my new PR